mod-source-record-storage-server/src/main/java/org/folio/verticle/MarcIndexersVersionDeletionVerticle.java

```diff
@@ -25,55 +35,122 @@ public class MarcIndexersVersionDeletionVerticle extends AbstractVerticle {
 
   private static final Logger LOGGER = LogManager.getLogger();
 
-  private RecordDao recordDao;
-  private TenantDataProvider tenantDataProvider;
+  private static final ZoneId ZONE_ID = ZoneId.systemDefault();
 
-  @Autowired
-  public MarcIndexersVersionDeletionVerticle(RecordDao recordDao, TenantDataProvider tenantDataProvider) {
-    this.recordDao = recordDao;
-    this.tenantDataProvider = tenantDataProvider;
-  }
+  private final RecordDao recordDao;
+  private final TenantDataProvider tenantDataProvider;
+  private List<LocalTime> scheduleTimes;
 
   @Value("${srs.marcIndexers.delete.interval.seconds:1800}")
   private int interval;
 
+  @Value("${srs.marcIndexers.delete.plannedTime:00:00}")
+  private String plannedTime;
+
+  @Value("${srs.marcIndexers.delete.dirtyBatchSize:100000}")
+  private Integer dirtyBatchSize;
+
   private Future<Boolean> currentDeletion = Future.succeededFuture();
 
+  @Autowired
+  public MarcIndexersVersionDeletionVerticle(RecordDao recordDao, TenantDataProvider tenantDataProvider) {
+    this.recordDao = recordDao;
+    this.tenantDataProvider = tenantDataProvider;
+  }
+
   @Override
   public void start(Promise<Void> startFuture) {
-    vertx.setPeriodic(interval * 1000L, id -> {
-      if (currentDeletion.isComplete()) {
-        currentDeletion = deleteOldMarcIndexerVersions();
-      } else {
-        LOGGER.info("Previous marc_indexers old version deletion still ongoing");
-      }
-    });
+    LOGGER.info("Specified values: planned time: {}, interval: {}", plannedTime, interval);
+    long intervalMillis = interval * 1000L;
+
+    if (!"00:00".equals(plannedTime) && interval == 1800) {
```

Comment:
I think you should remove the default value from the planned time. If planned time is not set use the interval. The problem I see is that if the interval default is changed above, a developer might not know to come change `1800`.

Reply:
Fixed. Thank you